### ISSUE TITLE
[BUG] Doesn't work well with Stage Manager

### ISSUE THREAD

**Issue by Hazza42** [ejbills/DockDoor#299]
## Describe the bug
When stage manager is active, window previews are distorted and very low resolution.

## Steps to reproduce
- Enable Stage Manager.

## Expected behavior
For window previews to appear flat.

## Actual behavior
It seems window previews are being captured from the stage manager area on the left side of the screen where they are tilted.

## Screenshots
![IMG_0678](https://github.com/user-attachments/assets/e63b9f81-e2e0-4d03-b525-6dbb98500335)

## Environment
 - macOS version: [macOS 13.6.7 Ventura]
 - App version: [1.2.3]

**Comment by illavoluntas** [comment]
That’s indeed a strange behavior, and I suppose it hasn’t been seen on macOS 14, otherwise it would have been reported a long time ago…

**Comment by Hazza42** [comment]
Looks like it’s time to update MacOS then! Fingers crossed that fixed it

**Comment by Hazza42** [comment]
Perhaps the solution for now then is to just disable Stage Manager? Maybe the reason nobody has reported it yet is because most people just don’t use Stage Manager.

**Comment by illavoluntas** [comment]
> Perhaps the solution for now then is to just disable Stage Manager? Maybe the reason nobody has reported it yet is because most people just don’t use Stage Manager.

Yeah, I think you're right. That's what I would do waiting for this issue to be addressed.

**Comment by illavoluntas** [comment]
Thanks for reporting it :smile:, looks like you're the first to report (maybe not the first one to notice) this bug.

**Comment by ejbills** [comment]
Right, I'm taking an educated guess that there isn't really anything for DockDoor to account for here. DockDoor sees the window belongs to the app, grabs the preview available via the screen capture API, and returns it. Unfortunately, the window is just tiny-so it has a tiny image preview. 

This issue was discovered way back a few months ago when I was doing closed alpha testing when DockDoor barely worked.😆

I will keep this discussion open for now, perhaps projects like AltTab have discovered a method of getting around this.

**Comment by illavoluntas** [comment]
Well, isn’t Stage Manager a sort of alternative to DockDoor in itself? It doesn’t have all features but it’s very similar.

**Comment by Yann-R** [comment]
Hello @illavoluntas 

> I suppose someone who uses Stage Manager is less prone to open the Dock…

Not always. Personally I use both Stage Manager and the Dock, so I feel glad to discover DockDoor and its window previews after having used HyperDock for years with the same (main) aim at getting previews in the Dock.
(for me Stage Manager helps tidy up my windows automatically into stacks, but Dock previews help me find a specific Window when its not clearly visible with Stage Manager stacked windows)

Concerning the "disappointment" with windows that are minimized to Stage Manager that result in non rectangular and non aesthetic snapshots/previews:
- I understand this is a consequence of the system itself
- but wouldn't there be a possible smart behavior inspired by this: before being minimized as a sort of parallelogram in Stage Manager, the concerned window was on screen and rectangular. Couldn't it be possible to keep in DockDoor previews only the rectangular window snapshots got from the system, and not refresh the preview when DockDoor get something not rectangular so as to keep the older, but rectangular window snapshot?

Just an idea, maybe no hope?

P.S. This is related to enhancement requests https://github.com/ejbills/DockDoor/issues/459, https://github.com/ejbills/DockDoor/issues/593, https://github.com/ejbills/DockDoor/issues/609

**Comment by SJeffZhang** [comment]
@Yann-R I'm a user who shares the same needs as you, and I'm really saddened that this issue can't be resolved, even though I understand it's a macOS system limitation.
